surrogates/common_utils/utils.py

```diff
@@ -1,38 +1,73 @@
 ##==============================================================================
-## BHPTNRSur1dq1e4 : arXiv:2204.01972
+## BHPTNRSurrogate module
 ## Description : utility file for surrogate model
 ## Author : Tousif Islam, Aug 2022 [tislam@umassd.edu / tousifislam24@gmail.com]
 ##==============================================================================
 
 import numpy as np
 from gwtools import gwtools as _gwtools
 from gwtools.harmonics import sYlm as _sYlm
+from . import nr_calibration as nrcalib
+
+#----------------------------------------------------------------------------------------------------
+def amp_ph_to_comp(amp,phase):
+    """ Takes the amplitude and phase of the waveform and
+    computes the compose them together"""
+    
+    full_wf = amp*np.exp(1j*phase)
+    return full_wf
+
+#----------------------------------------------------------------------------------------------------
+def re_im_to_comp(re, im):
+    """ Takes the real and imaginary part of the waveform and
+    combine them to obtain the coorbital frame waveform"""
+    
+    full_wf = (re+1j*im)
+    return full_wf
+
+#----------------------------------------------------------------------------------------------------
+def coorbital_to_inertial(h_coorb):
+    """ Transform the coorbital frame wf into the inertial frame"""
+    
+    h_inertial = {}
+    # 22 mode is in inertial frame and HMs are in coorbital phase
+    for mode in h_coorb.keys():
+        (l,m)=mode
+        if mode==(2,2):
+            # compute orbital phase
+            orbital_phase = np.unwrap(np.angle(h_coorb[mode]))/2
+            h_inertial[mode] = h_coorb[mode]
+        else:
+            # transform HMs to inertial frame
+            h_inertial[mode] = h_coorb[mode]*np.exp(1j*m*np.array(orbital_phase))
+            
+    return h_inertial
 
 #---------------------------------------------------------------------------------------------------- 
 def geo_to_SI(t_geo, h_geo, M_tot, dist_mpc):
     """
     transforms the waveform from geomeric unit to physical unit
     given geoemtric time, geometric waveform, total mass M, distance dL
     """    
-    # Physical units
+    # set constants
     G=_gwtools.G
     MSUN_SI = _gwtools.MSUN_SI
     PC_SI = _gwtools.PC_SI
     C_SI = _gwtools.c
+    # Physical units
     M = M_tot * MSUN_SI
     dL = dist_mpc * PC_SI
     
-    # scaling of time and strain
+    # scaling of time
     t_SI = t_geo * (G*M/C_SI**3)
-    
+    # scaling of strain for all modes
     strain_geo_to_SI = (G*M/C_SI**3)/dL
     h_SI={}
     for mode in h_geo.keys():
         h_SI[(mode)] = np.array(h_geo[mode])*strain_geo_to_SI
     
     return t_SI, h_SI
 
-
 #---------------------------------------------------------------------------------------------------- 
 def phase_rotation(h, delta_orb_phase):
```

Comment:
maybe this routine should be moved to gwtools? Its now used both here and gwsurrogate. Its also generally useful so a stand-alone function could make sense

Reply:
Fine  to me